docs/DeveloperGuide.md

```diff
@@ -452,6 +453,59 @@ The sequence diagram below illustrates the interactions happening within the `Lo
 
 A similar execution scenario can be expected for setting a student's payment status as `unpaid`.
 
+### Add lesson feature
+
+#### Implementation
+
+This feature adds a lesson to TutorAid. A lesson consists of the lesson name, capacity, pricing, timing and student list.
+
+This feature implements the following operations:
+
+* `AddLessonCommand#execute()` - Adds a `Lesson` object to TutorAid.
+
+This feature is additionally facilitated by the following operations:
+
+* `TutorAidParser#parseCommand()` - Checks for the command word that is required for the addition of a lesson.
+* `AddCommandParser#parse()` - Checks for the command flag that specifies the addition of a lesson.
+* `AddLessonCommandParser#parse()` - Parses the arguments provided to create a `Lesson` object.
+
+To represent a lesson and its details, a `Lesson` class is introduced. Its class diagram can be seen below.
+
+<img src="images/LessonClassDiagram.png" width="500"/>
+
+Given below is an example usage scenario for adding a lesson to TutorAid, and how the command is executed.
+
+1. The user executes `add -l n/Maths 1 c/15` to add a lesson named `Maths 1` which has a capacity of `15` to TutorAid. `LogicManager#execute()` is executed, where the user input is passed into `TutorAidParser#parseCommand()`. 
+
+2. This in turn calls `AddCommandParser#parse()`, where the string `-l n/Maths 1 c/15` is passed in as the argument. Due to the command flag `-l` at the start of the argument, `AddLessonCommandParser#parse()` is called to parse the remaining parameters `n/Maths 1 c/15` into lesson details.
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** 
+At this point, if `AddLessonCommandParser#parse()` detects that no lesson name has been supplied, the command will fail 
+its execution and `ParseException` will be thrown.
+</div>
+
+3. Since the optional parameters (`price` and `timing`) are not provided in the command, a default argument (`""`)  is supplied for these parameters. Else, the original arguments (`Maths 1` and `15`) are used for the parameters `lessonName` and `capacity` respectively. These parameters are then used to create `LessonName`, `Price`, `Capacity` and `Timing` instances.
```

Comment:
Maybe you can switch the first and second sentence so that it flows better? (e.g. The original arguments (`Maths 1` and `15`) are used for the parameters `lessonName` and `capacity` respectively. Since the optional parameters (`price` and `timing`) are not provided in the command, a default argument (`""`)  is supplied for these parameters.)

Reply:
Yep I have switched them, thanks!